src/stubgen.py

```diff
@@ -81,11 +82,45 @@ class and repeatedly call ``.put()`` to register modules or contents within the
 ]
 # fmt: on
 
+# (name, desired_as_name) -> actual_as_name
+ImportDict = Dict[Tuple[Optional[str], Optional[str]], Optional[str]]
+# package_name -> ((name, desired_as_name) -> actual_as_name)
+PackagesDict = Dict[str, ImportDict]
+
+# (signature_str, doc_str, default_arg_1, default_arg_2, ...)
+# We could type as Tuple[Optional[str], Optional[str], *Tuple[Any, ...]]
+# but this is only valid in python >= 3.11
+NbSignature = Tuple[Optional[str], Optional[str]]
+NbGetterSetterSignature = Tuple[str, str]
+
+
```

Comment:
We can do:
```python
@typing.runtime_checkable
class NbObject(Protocol):
    __nb_signature__: Tuple[NbSignature, ...]
```

Which allows us to check if a variable has a type that contains all the required field (i.e. `__nb_signature__`) by `isinstance(some_nb_function, NbObject)`. I think this is very neat, but not sure if you have any uses of this.